lib/index.js

```diff
@@ -40,12 +40,12 @@ internals.implementation = function (server, options) {
               return reply(Boom.unauthorized(JSON.stringify(err), 'Token'));
             }
             else { // see: http://hapijs.com/tutorials/auth for validateFunc signature
-              options.validateFunc(decoded, request, function (err, valid) { // bring your own checks
+              options.validateFunc(decoded, request, function (err, valid, credentials) { // bring your own checks
                 if(err || !valid) {
                   return reply(Boom.unauthorized('Credentials were not validated', 'Token'));
                 }
                 else {
-                  return reply.continue({ credentials: decoded });
+                  return reply.continue({ credentials: credentials || decoded });
```

Comment:
@ssassi I can see how this might be useful. 
but you have added a conditional to the code and not added a test to the suite. 
Can you please create a test for this new functionality?
thanks! https://github.com/ideaq/hapi-auth-jwt2/blob/master/test/test.js#L114

Also we would need to add this to the documentation, not just the library. :wink: